src/GraphQL/Types/Scalars/FloatGraphType.cs

```diff
@@ -10,40 +10,22 @@ namespace GraphQL.Types
     public class FloatGraphType : ScalarGraphType
     {
         /// <inheritdoc/>
-        public override object? ParseLiteral(IValue value) => value switch
+        public override object? ParseLiteral(GraphQLValue value) => value switch
         {
-            FloatValue floatVal => floatVal.Value,
-            IntValue intVal => (double)intVal.Value,
-            LongValue longVal => (double)longVal.Value,
-            DecimalValue decVal => checked((double)decVal.Value),
-            BigIntValue bigIntVal => checked((double)bigIntVal.Value),
-            NullValue _ => null,
+            GraphQLIntValue x => Double.Parse(x.Value),
+            GraphQLFloatValue x => Double.Parse(x.Value),
```

Comment:
I think the code should be explicit about using the invariant number format style, which is required per the graphql specs.  (in all scalars where appropriate)

Reply:
Do not forget that we need to include a patch for floating point values (float/double):

> In .NET Core 3.0 and later, values that are too large to represent are rounded to PositiveInfinity or NegativeInfinity as required by the IEEE 754 specification. In prior versions, including .NET Framework, parsing a value that was too large to represent resulted in failure.

See the old code for a sample -- basically, I think we use TryParse and if failure then examine the first character -- if it's a `-` then NegativeInfinity, else Infinity.  This also means that `CanParseLiteral` can return `true` for any `GraphQLFloatValue`.

Reply:
I understand what you are talking about, but not sure that it is right.

Reply:
It conforms to IEEE 754, which is explicitly stated we should be supporting in the GraphQL specification for floating-point values.

https://spec.graphql.org/June2018/#sec-Float

Reply:
I pushed a fix. Actually behavior differs on NET48 (throws) and NET Core (can parse infinity).

Reply:
I mean it differs for parsing doubles and after fix behavior the same for each platform -  (+/-)infinity and Nan throw.

Reply:
Correct